### ISSUE TITLE
Fails to browse messages on MSK

### ISSUE THREAD

**Issue by jaehyeon-kim** [provectus/kafka-ui#3985]
### Issue submitter TODO list

- [X] I've looked up my issue in [FAQ](https://docs.kafka-ui.provectus.io/faq/common-problems)
- [X] I've searched for an already existing issues [here](https://github.com/provectus/kafka-ui/issues)
- [X] I've tried running `master`-labeled docker image and the issue still persists there
- [X] I'm running a supported version of the application which is listed [here](https://github.com/provectus/kafka-ui/blob/master/SECURITY.md)

### Describe the bug (actual behavior)

It fails to browse messages of a topic. It is published to MSK where the Kafka version is 2.8.1. I can see topic details and associating consumers but not able to browse messages. The details of the topic is shown below.

![topic-details](https://github.com/provectus/kafka-ui/assets/1023748/fe2e73ad-da8d-4561-83b6-de735ad6d452)

### Expected behavior

I should be able to browse messages.

### Your installation details

App version - [727f384](https://github.com/provectus/kafka-ui/commit/727f384)

Docker compose file

```
version: "3"

services:
  kafka-ui:
    image: provectuslabs/kafka-ui:master
    container_name: kafka-ui
    ports:
      - "8080:8080"
    networks:
      - kafkanet
    environment:
      AWS_ACCESS_KEY_ID: $AWS_ACCESS_KEY_ID
      AWS_SECRET_ACCESS_KEY: $AWS_SECRET_ACCESS_KEY
      AWS_SESSION_TOKEN: $AWS_SESSION_TOKEN
      KAFKA_CLUSTERS_0_NAME: msk
      KAFKA_CLUSTERS_0_BOOTSTRAPSERVERS: $BOOTSTRAP_SERVERS
      # KAFKA_CLUSTERS_0_DISABLELOGDIRSCOLLECTION: true # only for MSK serverless
      KAFKA_CLUSTERS_0_PROPERTIES_SECURITY_PROTOCOL: SASL_SSL
      KAFKA_CLUSTERS_0_PROPERTIES_SASL_MECHANISM: AWS_MSK_IAM
      KAFKA_CLUSTERS_0_PROPERTIES_SASL_CLIENT_CALLBACK_HANDLER_CLASS: software.amazon.msk.auth.iam.IAMClientCallbackHandler
      KAFKA_CLUSTERS_0_PROPERTIES_SASL_JAAS_CONFIG: software.amazon.msk.auth.iam.IAMLoginModule required;

networks:
  kafkanet:
    name: kafka-network
```

### Steps to reproduce

1. Create a MSK cluster (version 2.8.1)
2. Deploy a source connector with the following properties listed at the end. The source connector is created by [MSK Data Generator](https://github.com/awslabs/amazon-msk-data-generator)
3. Run the docker compose file

```
    # connector configuration
    "connector.class"                = "com.amazonaws.mskdatagen.GeneratorSourceConnector",
    "tasks.max"                      = "1",
    "key.converter"                  = "org.apache.kafka.connect.storage.StringConverter",
    "key.converter.schemas.enable"   = false,
    "value.converter"                = "org.apache.kafka.connect.json.JsonConverter",
    "value.converter.schemas.enable" = false,
    # msk data generator configuration
    "genkp.order.with"              = "to-replace",
    "genv.order.order_id.with"      = "#{Internet.uuid}",
    "genv.order.product_id.with"    = "#{Code.isbn10}",
    "genv.order.quantity.with"      = "#{number.number_between '1','5'}",
    "genv.order.customer_id.with"   = "#{number.number_between '100','199'}",
    "genv.order.customer_name.with" = "#{Name.full_name}",
    "global.throttle.ms"            = "500",
    "global.history.records.max"    = "1000",
    # single message transforms
    "transforms"                            = "copyIdToKey,extractKeyFromStruct,cast",
    "transforms.copyIdToKey.type"           = "org.apache.kafka.connect.transforms.ValueToKey",
    "transforms.copyIdToKey.fields"         = "order_id",
    "transforms.extractKeyFromStruct.type"  = "org.apache.kafka.connect.transforms.ExtractField$Key",
    "transforms.extractKeyFromStruct.field" = "order_id",
    "transforms.cast.type"                  = "org.apache.kafka.connect.transforms.Cast$Value",
    "transforms.cast.spec"                  = "quantity:int8"
```

### Logs

Below shows the logs. It indicates `0 records polled`.

```
2023-06-29 05:41:40,918 INFO  [boundedElastic-7] o.a.k.c.u.AppInfoParser: Kafka version: 3.3.1
2023-06-29 05:41:40,918 INFO  [boundedElastic-7] o.a.k.c.u.AppInfoParser: Kafka commitId: e23c59d00e687ff5
2023-06-29 05:41:40,918 INFO  [boundedElastic-7] o.a.k.c.u.AppInfoParser: Kafka startTimeMs: 1688017300918
2023-06-29 05:41:43,880 INFO  [boundedElastic-7] o.a.k.c.Metadata: [Consumer clientId=kafka-ui-consumer-1688017300912, groupId=null] Cluster ID: vlkbd43bTb2JRIvX3phegA
2023-06-29 05:41:47,396 INFO  [boundedElastic-7] o.a.k.c.Metadata: [Consumer clientId=kafka-ui-consumer-1688017300912, groupId=null] Resetting the last seen epoch of partition order-0 to 0 since the associated topicId changed from null to IfdlNP7BTP2KsEe_mIqT7w
2023-06-29 05:41:47,396 INFO  [boundedElastic-7] o.a.k.c.Metadata: [Consumer clientId=kafka-ui-consumer-1688017300912, groupId=null] Resetting the last seen epoch of partition order-2 to 0 since the associated topicId changed from null to IfdlNP7BTP2KsEe_mIqT7w
2023-06-29 05:41:47,396 INFO  [boundedElastic-7] o.a.k.c.Metadata: [Consumer clientId=kafka-ui-consumer-1688017300912, groupId=null] Resetting the last seen epoch of partition order-1 to 0 since the associated topicId changed from null to IfdlNP7BTP2KsEe_mIqT7w
2023-06-29 05:41:51,200 INFO  [boundedElastic-7] o.a.k.c.c.KafkaConsumer: [Consumer clientId=kafka-ui-consumer-1688017300912, groupId=null] Assigned to partition(s): order-2, order-0, order-1
2023-06-29 05:41:51,201 INFO  [boundedElastic-7] o.a.k.c.c.KafkaConsumer: [Consumer clientId=kafka-ui-consumer-1688017300912, groupId=null] Seeking to offset 0 for partition order-2
2023-06-29 05:41:51,201 INFO  [boundedElastic-7] o.a.k.c.c.KafkaConsumer: [Consumer clientId=kafka-ui-consumer-1688017300912, groupId=null] Seeking to offset 0 for partition order-0
2023-06-29 05:41:51,201 INFO  [boundedElastic-7] o.a.k.c.c.KafkaConsumer: [Consumer clientId=kafka-ui-consumer-1688017300912, groupId=null] Seeking to offset 0 for partition order-1
2023-06-29 05:41:52,203 DEBUG [boundedElastic-7] c.p.k.u.e.ForwardRecordEmitter: 0 records polled
2023-06-29 05:41:53,206 DEBUG [boundedElastic-7] c.p.k.u.e.ForwardRecordEmitter: 0 records polled
2023-06-29 05:41:54,211 DEBUG [boundedElastic-7] c.p.k.u.e.ForwardRecordEmitter: 0 records polled
2023-06-29 05:41:54,212 DEBUG [boundedElastic-7] c.p.k.u.e.ForwardRecordEmitter: Polling finished
```

[log.log](https://github.com/provectus/kafka-ui/files/11902383/log.log)

### Additional context

I checked with the [kpow CE](https://docs.kpow.io/ce/) and there is not problem to browse messages.

**Comment by Haarolean** [comment]
Hi, most likely, it's related to network.

I recommend tuning polling configuration, documented [here](https://docs.kafka-ui.provectus.io/configuration/misc-configuration-properties)
`kafka.polling.pollTimeoutMs`, `kafka.polling.partitionPollTimeout`, `kafka.polling.noDataEmptyPolls`

Let us know how it goes

**Comment by krumft** [comment]
Have you by chance enabled compression on your MSK topics? I think KafkaUI does not yet support browsing such topics: https://github.com/provectus/kafka-ui/issues/4005

**Comment by jaehyeon-kim** [comment]
@Haarolean 

Thanks for the suggestion. However it didn't have luck after updating those config values. Let me try again later.

@krumft 

No, messages are not compressed.

**Comment by Haarolean** [comment]
Most likely it's related to polling timeouts. We've implemented one more set of changes aimed to fix such issues within #4088. We'll release 0.7.2 this week, please check it out once it's out and let us know how it works.
